### DISCUSSION TITLE
How to add Tuya device?

### DISCUSSION THREAD

**Discussion by ArJay60** [Koenkk/zigbee2mqtt#5388] [General]
On the how to page to support Tuya devices common.js is mentioned. Where did this file go in the latest development version? I want to add new datapoints. Is the How-to-page still correct?

Excerpt from page (https://www.zigbee2mqtt.io/how_tos/how_to_support_new_tuya_devices.html):

5. Adding your first data point #
Let’s assume we want to add the heating setpoint of the Saswell thermostat first.

First add the saswellHeatingSetpoint data point to TuyaDataPoints in node_modules/zigbee-herdsman-converters/convertes/**common.js** with value 103.

**Comment by Koenkk** [comment]
You can find it in `zigbee-herdsman-converters/lib/tuya.js`
